### PR TITLE
Added some more types and fixed some bugs

### PR DESCRIPTION
**Pull request by Chafficui**
Added types:
- Int64
- Int8

Fixed bugs:
- negative reading lengths

### CODE REVIEW

File: GvasFormat/Serialization/BinaryReaderEx.cs [hunk 1]
```diff
@@ -16,9 +16,15 @@ public static string ReadUEString(this BinaryReader reader)
             if (length == 0)
                 return null;
 
+            if (length < 0)
```
Comment (13xforever) [inline GvasFormat/Serialization/BinaryReaderEx.cs:19]: this doesn't make sense; where is this atually used? as far as I can remember, ue4 serialization is forward-only, so if it ever needs to seek back, it's probably a bug or some unimplemented feature

**Comment by 13xforever** [comment]
thank you for contribution, though this project is pretty much dead until the day I'll need it again

File: GvasFormat/Serialization/BinaryReaderEx.cs (hunk 1 above)
Comment (Chafficui) [reply GvasFormat/Serialization/BinaryReaderEx.cs:19]: Yeah I had this problem with F1 Manager 2022, but you may be right ... It may be a bug

Comment (13xforever) [reply GvasFormat/Serialization/BinaryReaderEx.cs:19]: I'll pull other changes, but this looks so wrong and this is for strings too, you'll get garbage in the beginning and will probably read into garbage at the end, so I'd ask to updata the pr without this change

Comment (Chafficui) [reply GvasFormat/Serialization/BinaryReaderEx.cs]: Updated it

**Comment by 13xforever** [comment]
and it still changed the csproj gdi github

**Pull request merged**